src/Engine/GameAction.java

```diff
@@ -828,6 +831,100 @@ public UnitActionType getUnitActionType()
     }
   } // ~ResupplyAction
 
+  // ===========  RepairUnitAction  ===============================
+  public static class RepairUnitAction implements GameAction
+  {
+    private Path movePath;
+    private XYCoord startCoord;
+    private XYCoord moveCoord;
+    private XYCoord repairCoord;
+
+    public RepairUnitAction(XYCoord startLoc, Path path, int targetX, int targetY)
+    {
+      this(startLoc, path, new XYCoord(targetX, targetY));
+    }
+
+    public RepairUnitAction(XYCoord startLoc, Path path, XYCoord repairLoc)
+    {
+      movePath = path;
+      startCoord = startLoc;
+      repairCoord = repairLoc;
+      if( null != path && (path.getEnd() != null) )
+      {
+        moveCoord = new XYCoord(movePath.getEnd().x, movePath.getEnd().y);
+      }
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      // Repair actions consist of
+      //   MOVE
+      //   HEAL
+      //   REPAIR
+      GameEventQueue repairEvents = new GameEventQueue();
+
+      boolean isValid = true;
+      Unit benefactor = null;
+      Unit beneficiary = null;
+
+      if( (null != gameMap) && (null != startCoord) && (null != repairCoord) &&
+          gameMap.isLocationValid(startCoord) && gameMap.isLocationValid(repairCoord) )
+      {
+        benefactor = gameMap.getLocation(startCoord).getResident();
+        beneficiary = gameMap.getLocation(repairCoord).getResident();
+        isValid &= benefactor != null && !benefactor.isTurnOver;
+        isValid &= (null != beneficiary) && !benefactor.CO.isEnemy(beneficiary.CO);
+        isValid &= (null != gameMap) && (gameMap.isLocationValid(repairCoord));
+        isValid &= (movePath != null) && (movePath.getPathLength() > 0);
+      }
+      else
+        isValid = false;
+
+      if( isValid )
+      {
+        Location moveLocation = gameMap.getLocation(moveCoord);
+        isValid &= (null == moveLocation.getResident()) || (benefactor == moveLocation.getResident());
+      }
+
+      if( isValid )
+      {
+        if( Utils.enqueueMoveEvent(gameMap, benefactor, movePath, repairEvents) )
+        {
+          // No surprises in the fog.
+          repairEvents.add(new HealUnitEvent(beneficiary, true));
```

Comment:
Hah, I've had a `UnitHealEvent` locally for a while, but it never got added (I think because you objected to separating funds transfer from healing). Ah well.

Reply:
Is there a good reason to separate them? Seems like it would add complexity for little gain.

Reply:
If we combine them, I think `UnitRepairEvent` makes more sense than `Heal`. Separating them could make sense because it allows them to happen independently, and allows tracking/listening to cases where one happens but not the other.

Reply:
With `Heal`, I was following your convention with `CreateUnitEvent`. I don't think of "this costs money" with either "heal" or "create" unit, but adding the option in the constructor would make it plenty clear.

I suppose we could make a "MoneyChange(Delta?)Event" that would give or remove cash from a player, but it feels like that's the only use case where there's just money changes happening and nothing else. Except it could be a "TurnInitEvent" just as well.

As for tracking/listening, I think it makes more sense to add whether there was a money cost in the published values. One more optional if-check is less added complexity than at least a new class and definitely at least one more function.
Plus, splitting them into separate events means you no longer know for sure *why* the money is changing.

Reply:
Hmm. Yeah, with CO powers that spawn units for no cost, we may have to tweak `CreateUnitEvent` or make a new one. Ah well, out of scope for this PR.
I don't object to this approach, I was just amused that you also created a healing event.